### ISSUE TITLE
How to Install?

### ISSUE THREAD

**Issue by FunnySunnyKitten** [MikeMcShaffry/kflipcamp#14]
I've got the files downloaded, but I'm confused about how to make it work with Discord. Is there any way you can make a guide on how to install it and make Discord read MediaMonkey's data?

**Comment by MikeMcShaffry** [comment]
Howdy there!

Discord won't read MediaMonkey's data directly, unfortunately....you have
to go about it in a really goofy way.

In the website config.json you set your Discord channel information, under
the "otto" section:

  "otto": {
    "enabled": true,
    "token": "BLAH",
    "prefix": "!",
    "listener_channel_id": "CHANNEL_ID_TO_POST_NOW_PLAYING",
    "engineering_channel_id": "CHANNEL_ID_TO_POST_OTTO_ACTIVITY_LOGS",
    "use_nowplayingfile": false,
    "nowplayingfile": "C:/temp/MediaMonkey_NowPlaying.txt"
  },

MediaMonkey doesn't stream to IceCast on its own - KFLIP happens to use
RocketBroadcaster, but you can also use apps like Mixx or BUTT. It is up to
those applications to add the right title information to the current song -
in the format of

*Artist* - *Song* off *Album*

Since we use MediaMonkey, we have a little plugin (kflip utilities) that
runs when the song changes and writes this information to a local file.

We licensed Rocket Broadcaster Pro which can look at that file and add the
title information to the stream it sends up to IceCast.

The website service polls the IceCast service and detects any change to the
song title, and if Otto is configured right it sends a message to Discord
"<dj> is playing <artist> - <song> off <album>"

It is a bit tricky to get installed because there are a lot of moving
parts, but if you are using MediaMonkey and RocketBroadcaster I can send
you some screenshots on how we got everything working.

**Comment by FunnySunnyKitten** [comment]
Yeah that would be really helpful if you could provide some images for what I'm supposed to do. I'm kind of a noob at this kind of stuff so it would help a lot. Where is the website config.json located?

**Comment by MikeMcShaffry** [comment]
Look in public_html/config.json.example - you would copy that file to
config.json, and replace all the values with yours.

A couple of questions:
- What is your experience level setting up web services? (I ask because
that will help me understand how detailed my other answers need to be)
- How are you setting up the webserver and IceCast server? Are you using a
paid service or rolling your own on something like AWS?
- What program do you use to stream to IceCast??
